# HTRC Extracted Features 2.5



In [1]:
!pip install htrc-feature-reader

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.4/58.4 kB 4.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 160.3/160.3 kB 12.7 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.8/1.8 MB 59.2 MB/s eta 0:00:00
  Created wheel for htrc-feature-reader: filename=htrc_feature_reader-2.0.7-py3-none-any.whl size=40405 sha256=4bcf4016d811a588afda558c551796c0c018cacd55b1f672c2b68b6d8bfb2e72
  Stored in directory: /root/.cache/pip/wheels/7e/09/8e/d5b37ad3e138bf601d53aeaef0d071416c0468d1d3b15ec656
Successfully built htrc-feature-reader


In [2]:
from google.colab import files
uploaded = files.upload()

Saving list.csv to list.csv


#Generate Download Paths

In [3]:
import csv
from htrc_features import utils
def normalize_htid(htid):
    # Fix CSV format: "ark_ 13960 xxx" -> "ark:/13960/xxx"
    if 'ark_' in htid:
        htid = htid.replace('ark_ ', 'ark:/')
        htid = htid.replace(' ', '/')
    else:
        htid = htid.replace(' ', '')
    return htid
# Generate paths
paths = []
with open('list.csv', 'r') as f:
    reader = csv.DictReader(f)
    for row in reader:
        htid = normalize_htid(row['htid'])
        path = utils.id_to_rsync(htid, format='stubbytree')
        paths.append(path)
# Save to file
with open('file_paths.txt', 'w') as f:
    for path in paths:
        f.write(path + '\n')
print(f"✓ Generated {len(paths)} paths")

✓ Generated 120 paths


## Download All Extracted Features
##### Download 120 files using rsync.

In [4]:
!mkdir -p ef25_data
!rsync -av --files-from=file_paths.txt data.analytics.hathitrust.org::features-2025.04/ ef25_data/

[htrc-data] Welcome to the HathiTrust Research Center rsync server.

receiving file list ... done
dul1/
dul1/a+30233/
dul1/a+30233/dul1.ark+=13960=t27b3zv3b.json.bz2
dul1/a+3023c7/
dul1/a+3023c7/dul1.ark+=13960=s2st3fdcwc7.json.bz2
dul1/a+30305/
dul1/a+30305/dul1.ark+=13960=t3f00205w.json.bz2
emu/
emu/0011/
emu/0011/emu.010001132199.json.bz2
emu/0024/
emu/0024/emu.010001297436.json.bz2
emu/0071/
emu/0071/emu.010000709155.json.bz2
hvd/
hvd/34219/
hvd/34219/hvd.32044020016994.json.bz2
hvd/34342/
hvd/34342/hvd.32044134446327.json.bz2
hvd/34512/
hvd/34512/hvd.32044050812528.json.bz2
hvd/34896/
hvd/34896/hvd.32044080899966.json.bz2
hvd/hi/
hvd/hi/hvd.hn1i81.json.bz2
hvd/hn/
hvd/hn/hvd.hn1nyx.json.bz2
ien/
ien/35352/
ien/35352/ien.35556034454629.json.bz2
inu/
inu/30005/
inu/30005/inu.30000107605259.json.bz2
inu/30085/
inu/30085/inu.30000005083658.json.bz2
inu/30133/
inu/30133/inu.30000111037739.json.bz2
inu/30167/
inu/30167/inu.30000111363671.json.bz2
inu/30203/
inu/30203/inu.30000022309136.

#### Zip Result

In [5]:
!zip -r ef25_data.zip ef25_data/
files.download('ef25_data.zip')


  adding: ef25_data/ (stored 0%)
  adding: ef25_data/umn/ (stored 0%)
  adding: ef25_data/umn/35011/ (stored 0%)
  adding: ef25_data/umn/35011/umn.31951d02915219o.json.bz2 (deflated 0%)
  adding: ef25_data/txu/ (stored 0%)
  adding: ef25_data/txu/01020/ (stored 0%)
  adding: ef25_data/txu/01020/txu.059173018279060.json.bz2 (deflated 0%)
  adding: ef25_data/txu/01022/ (stored 0%)
  adding: ef25_data/txu/01022/txu.059173014228227.json.bz2 (stored 0%)
  adding: ef25_data/txu/01063/ (stored 0%)
  adding: ef25_data/txu/01063/txu.059173009680328.json.bz2 (deflated 0%)
  adding: ef25_data/txu/01028/ (stored 0%)
  adding: ef25_data/txu/01028/txu.059173010296815.json.bz2 (stored 0%)
  adding: ef25_data/txu/01019/ (stored 0%)
  adding: ef25_data/txu/01019/txu.059173010198907.json.bz2 (stored 0%)
  adding: ef25_data/emu/ (stored 0%)
  adding: ef25_data/emu/0024/ (stored 0%)
  adding: ef25_data/emu/0024/emu.010001297436.json.bz2 (deflated 0%)
  adding: ef25_data/emu/0071/ (stored 0%)
  adding: ef2

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>